scripts/west_commands/ncs-provision.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from pathlib import Path
+import re
+import sys
+import subprocess
+from cryptography.hazmat.primitives.serialization import load_pem_private_key
+from west.commands import WestCommand
+
+nrf54l15_key_slots = [226, 228, 230]
+
+
+class NcsProvision(WestCommand):
+    def __init__(self):
+        super().__init__(
+            "ncs-provision",
+            "NCS provision",
+            "NCS provision utility tool.",
+        )
+
+    def do_add_parser(self, parser_adder):
+        parser = parser_adder.add_parser(
+            self.name, help=self.help, description=self.description
+        )
+
+        subparsers = parser.add_subparsers(
+            dest="command"
+        )
+        upload_parser = subparsers.add_parser("upload", help="Send to KMU")
+        upload_parser.add_argument(
+            "-k", "--key", type=Path, action='append', dest="keys",
+            help="Input .pem file with ED25519 private key"
+        )
+        upload_parser.add_argument("-s", "--soc", type=str, help="SoC",
+                                   choices=["nrf54l15"],  required=True)
+
+        return parser
+
+    def do_run(self, args, unknown_args):
+        if args.command == "upload":
+            if args.soc == "nrf54l15":
+                if len(args.keys) > len(nrf54l15_key_slots):
+                    sys.exit(
+                        "Error: requested upload of more keys than there are designated slots.")
+                slot = 0
+                for keyfile in args.keys:
+                    with open(keyfile, 'rb') as f:
+                        priv_key = load_pem_private_key(f.read(), password=None)
+                    pub_key = priv_key.public_key()
+                    print(pub_key.public_bytes_raw().hex())
+                    nrfprovision = subprocess.run(
+                       ["nrfprovision",
+                        "provision",
+                        "-r",
+                        "REVOKED",
+                        "-v",
+                        pub_key.public_bytes_raw().hex(),
+                        "-m",
+                        "0x10ba0030",
+                        "-i",
+                        str(nrf54l15_key_slots[slot]),
+                        "-a",
+                        "ED25519",
+                        "-d",
+                        "0x20000000",
+                        "--verify"],
+                       stderr=subprocess.PIPE,
+                       text=True)
+                    stderr = nrfprovision.stderr
+                    print(stderr)
+                    if re.search('fail', stderr) or nrfprovision.returncode:
+                        sys.exit("\033[1;31mUploading failed!")
```

Comment:
This colorizing may not work for Windows and in Jenkins CI, and with some terminal. Also you should reset the color at the end, so user won't end up with read text in his terminal. 
It is better just to remove the color from here.

Reply:
you should probably use `is_tty()` or similar and also perhaps colorama or a similar library. But non-blocking for me.

Reply:
or just use / copy code from other west-commands, you can use `from west import log` and print messages with `log.msg`, `log.dbg`, and `log.die` that also calls sys.exit

Reply:
ok, let's keep it simple for now.